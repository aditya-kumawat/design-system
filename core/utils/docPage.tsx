// @ts-nocheck

import * as React from 'react';
import { Title, Subtitle, Heading, Subheading, Props, Description, Preview, Source, PropsProps } from '@storybook/addon-docs/blocks';
import { renderToStaticMarkup } from 'react-dom/server';
import reactElementToJSXString from 'react-element-to-jsx-string';
import { html as beautifyHTML } from 'js-beautify';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { docco, dark, a11yDark, githubGist, dracula, vs2015 } from 'react-syntax-highlighter/dist/esm/styles/hljs';
import * as DS from '@/';
import { Button, Card, TabsWrapper, Tab } from '@/';
import vsDark from 'prism-react-renderer/themes/vsDark';

import {
  LiveProvider,
  LiveEditor,
  LiveError,
  LivePreview,
  withLive
} from 'react-live';

export interface Example {
  title: string;
  description?: string;
  imports: string[];
  component: React.ReactNode;
}

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
  customCode?: string;
}

const beautifyHTMLOptions = {
  indent_size: 2,
  wrap_line_length: 0,
  preserve_newlines: true,
  jslint_happy: true,
  end_with_newline: false,
  indent_inner_html: true
};

const beautifyJSXOptions = {
  ...beautifyHTMLOptions,
  e4x: true
};

const JSXtoStringOptions = {
  filterProps: (val: any, key: string) => {
    if (!val) return false;
    if (val && val.name === 'actionHandler') return false;
    // if(val && typeof val === 'function') return false;
    return true;
  },
  showFunctions: true,
  functionValue: _fn => _ => { },
  // maxInlineAttributesLineLength: 10
};

const copyCode = (val: string) => navigator.clipboard.writeText(val);

const importsToStr = (imports: string[]): string => `import { ${imports.join(', ')} } from 'design-system';`;

const CopyComp = props => {
  const { onClick } = props;
  return (
    <div
      style={{
        position: 'absolute',
        right: '0',
        top: '0',
        zIndex: 10
      }}
    >
      <Button size="small" style={{ borderRadius: '0' }} appearance="basic" onClick={onClick}>Copy</Button>
    </div>
  );
};

const renderCodeBlock = (val: string) => (
  <pre style={{ position: 'relative', fontFamily: 'monospace', fontSize: '13px', background: '#f8f8f8' }}>
    <CopyComp onClick={() => copyCode(val)} />
    <SyntaxHighlighter language="javascript" style={vs2015} showLineNumbers={true}>
      {val}
    </SyntaxHighlighter>
  </pre>
);

const StoryComp = props => {
  const {
    customCode
  } = props;

  const { storyId } = __STORYBOOK_STORY_STORE__.getSelection();
  const story = __STORYBOOK_STORY_STORE__.fromId(storyId);
  const comp = story.storyFn().props.children.props.children.props.children;
  const sp = story.parameters;
  const imports = [sp.component, ...(sp.subcomponents ? Object.values(sp.subcomponents) : [])];

  const importString = `// ${importsToStr(imports.map(i => i.displayName))}`;
  const jsx = customCode ? customCode : `${beautifyHTML(reactElementToJSXString(comp, JSXtoStringOptions), beautifyJSXOptions)}`;
  const html = beautifyHTML(renderToStaticMarkup(comp), beautifyHTMLOptions);

  const [activeTab, setActiveTab] = React.useState<number>(0);
  const [jsxCode, setJsxCode] = React.useState<string>(`${importString}\n\n${jsx}`);
  const [htmlCode, setHtmlCode] = React.useState<string>(`${html}`);

  const importScope = imports.reduce((out, curr) => {
    out[curr.displayName] = curr;
    return out;
  }, {});

  const TabsWrap = withLive(({ live }) => {
    const {
      error,
      element: Element,
    } = live;

    React.useEffect(() => {
      if (!error) {
        const editor = document.querySelector('.npm__react-simple-code-editor__textarea');

        if (editor) {
          const jsxValue = editor.value;
          setJsxCode(jsxValue);

          try {
            const htmlValue = beautifyHTML(renderToStaticMarkup(<Element />), beautifyHTMLOptions);
            setHtmlCode(htmlValue);
          } catch (e) { }
        }
      }
    }, [live]);

    return (
      <></>
    );
  });

  return (
    <>
      <Card
        shadow="light"
      >
        <LiveProvider code={jsxCode} scope={{ ...DS, ...importScope }}>
          <Preview
            style={{
              marginTop: '0px',
              marginBottom: '0px'
            }}
            withSource="none"
            withToolbar={true}
          >
            <LivePreview />
          </Preview>
          <TabsWrap />
          <TabsWrapper
            activeTab={activeTab}
            onTabChange={tab => setActiveTab(tab)}
          >
            <Tab label={'React'}>
              <div style={{ position: 'relative' }}>
                <CopyComp
                  onClick={() => {
                    const editor = document.querySelector('.npm__react-simple-code-editor__textarea');
                    if (editor) copyCode(editor.value);
                  }}
                />
                <LiveEditor theme={vsDark} />
                <LiveError />
              </div>
            </Tab>
            <Tab label={'HTML'}>
              {renderCodeBlock(htmlCode)}
            </Tab>
          </TabsWrapper>
        </LiveProvider>
      </Card>
    </>
  );
};

export const docPage = () => {
  const { storyId } = __STORYBOOK_STORY_STORE__.getSelection();
  const story = __STORYBOOK_STORY_STORE__.fromId(storyId);
  const comp = story.storyFn().props.children.props.children.props.children;
  const sp = story.parameters;

  const {
    title,
    description,
    props: propsAttr,
    customCode,
    noStory
  } = sp.docs.docPage;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <br />

      <Description>
        {description}
      </Description>
      <br />

      {!noStory && (
        <>
          <Heading>Story</Heading>
          <StoryComp key={storyId} customCode={customCode} />
          <br />
        </>
      )}

      <Heading>PropTable</Heading>
      <Props {...propsAttr} />
    </>
  );
};

export default docPage;
