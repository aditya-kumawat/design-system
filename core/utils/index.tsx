import * as positionHelper from './positionHelper';
import PopperWrapper from './PopperWrapper';

export { positionHelper,  PopperWrapper };
export * from './css';
