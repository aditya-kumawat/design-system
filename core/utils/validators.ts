export const dateValidator = (format: string, val: string): boolean => {
  const validate = (date: number, month: number, year: number): boolean => {
    // Check the ranges of month and year
    if (year < 1000 || year > 3000 || month == 0 || month > 12)
      return false;

    var monthLength = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];

    // Adjust for leap years
    if (year % 400 == 0 || (year % 100 != 0 && year % 4 == 0))
      monthLength[1] = 29;

    // Check the range of the day
    return date > 0 && date <= monthLength[month - 1];
  }

  switch (format) {
    case 'dd/mm/yy':
      var p = val.split('/');
      var date = +p[0];
      var month = +p[1];
      var year = +p[2];
      return validate(date, month, year);

    case 'mm/dd/yy':
      var p = val.split('/');
      var date = +p[1];
      var month = +p[0];
      var year = +p[2];
      return validate(date, month, year);

    case 'yy/mm/dd':
      var p = val.split('/');
      var date = +p[2];
      var month = +p[1];
      var year = +p[0];
      return validate(date, month, year);

    case 'dd-mm-yy':
      var p = val.split('-');
      var date = +p[0];
      var month = +p[1];
      var year = +p[2];
      return validate(date, month, year);

    case 'mm-dd-yy':
      var p = val.split('-');
      var date = +p[1];
      var month = +p[0];
      var year = +p[2];
      return validate(date, month, year);

    case 'yy-mm-dd':
      var p = val.split('-');
      var date = +p[2];
      var month = +p[1];
      var year = +p[0];
      return validate(date, month, year);

    default:
      return false;
  }
}

const e: Record<string, any> = {
  date: dateValidator
}
export default e;