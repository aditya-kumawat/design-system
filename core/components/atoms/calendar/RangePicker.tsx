import * as React from 'react';
import Calendar from './Calendar';
import { View, Day, DateType, DateFormat } from './types';
import Text from '@/components/atoms/text';
import Popover from '@/components/molecules/popover';
import InputMask from '@/components/molecules/inputMask';
import { getDateInfo, convertToDate } from './utility';
import validators from '@/utils/validators';

export interface IRangePickerProps {
  onRangeChange?: (startDate?: Date, endDate?: Date) => void;
  monthsInView?: number;
  jumpView?: boolean;
  firstDayOfWeek?: Day;
  view?: View;
  disabledBefore?: DateType;
  disabledAfter?: DateType;
  startDate?: DateType;
  endDate?: DateType;
  rangeLimit?: number;
  yearNav?: number;
  monthNav?: number;
  withInput?: boolean;
  inputFormat?: DateFormat;
  outputFormat?: DateFormat;
  rangeSeparator?: string;
}

export const RangePicker: React.FunctionComponent<IRangePickerProps> = props => {
  const {
    startDate: startDateProp,
    endDate: endDateProp,
    withInput = false,
    inputFormat = 'mm/dd/yy',
    outputFormat = 'mm/dd/yy',
    rangeSeparator = ' - ',
    onRangeChange,
    ...rest
  } = props;

  const [startDate, setStartDate] = React.useState<Date | undefined>(startDateProp ? convertToDate(startDateProp) : undefined);
  const [endDate, setEndDate] = React.useState<Date | undefined>(endDateProp ? convertToDate(endDateProp) : undefined);

  // React.useEffect(() => {
  //   console.log(startDate, endDate);
  // }, [startDate, endDate])

  if (withInput) {
    const translateToString = (format: string, d: Date): string => {
      const {
        year,
        month,
        date
      } = getDateInfo(d);

      const separator = format.includes('/') ? '/' : '-';
      const val = format.split(separator).reduce((out, curr) => {
        switch (curr) {
          case 'mm':
            out += (month < 10 && '0') + (month + 1);
            break;
          case 'yy':
            out += year;
            break;
          case 'dd':
            out += (date < 10 && '0') + date;
            break;
        }
        return out;
      }, '');

      return val;
    }

    const translateToDate = (format: string, val: string): Date | undefined => {
      if (validators.date(inputFormat, val)) {
        const separator = format.includes('/') ? '/' : '-';

        let year: number = -1,
          month: number = -1,
          date: number = -1;
        const v = val.split(separator);
        format.split(separator).forEach((f, i) => {
          switch (f) {
            case 'mm':
              month = +v[i] - 1;
              break;
            case 'yy':
              year = +v[i];
              break;
            case 'dd':
              date = +v[i];
              break;
          }
        });
        const d = convertToDate({ year, month, date });
        return d;
      } else {
        return undefined;
      }
    }

    const onChangeHandler = (_e: React.ChangeEvent<HTMLInputElement>, val?: string) => {
      const placeholderChar = '_';
      if (val && !val.includes(placeholderChar)) {
        const d = translateToDate(inputFormat, val);
        if (d) {
          if (!startDate) setStartDate(d);
          else setEndDate(d);
        }
      }
    }

    const onClearHandler = (type: string) => {
      if (type === 'start') setStartDate(undefined);
      if (type === 'end') setEndDate(undefined);
    }

    const onRangeChangeHandler = (sDate?: Date, eDate?: Date) => {
      if (sDate) setStartDate(sDate);
      if (eDate) setEndDate(eDate);
      if (onRangeChange) {
        onRangeChange(startDate, endDate);
      }
    }

    // const translateToInputValue = (format: string, sDate?: Date, eDate?: Date) => {
    //   let sVal = '',
    //     eVal = '';
    //   if(sDate) sVal = translateToString(format, sDate);
    //   if(eDate) eVal = translateToString(format, eDate);

    //   return sVal + rangeSeparator + eVal;
    // }

    const trigger = (
      <div className='RangePicker-input'>
        <InputMask
          name={"datepicker"}
          mask={{ name: 'date', type: inputFormat }}
          validator={{ name: 'date', type: inputFormat }}
          placeholder={inputFormat}
          value={startDate ? translateToString(inputFormat, startDate) : ''}
          onChange={onChangeHandler}
          onClear={() => onClearHandler('start')}
        />
        <div className="RangePicker-rangeSeparator">
          <Text>{rangeSeparator}</Text>
        </div>
        <InputMask
          name={"datepicker"}
          mask={{ name: 'date', type: inputFormat }}
          validator={{ name: 'date', type: inputFormat }}
          placeholder={inputFormat}
          value={endDate ? translateToString(inputFormat, endDate) : ''}
          onChange={onChangeHandler}
          onClear={() => onClearHandler('end')}
        />
      </div>
    );

    return (
      <Popover
        trigger={trigger}
        position='bottom-start'
        on='click'
        appendToBody={true}
      >
        <Calendar
          {...rest}
          rangePicker={true}
          startDate={startDate}
          endDate={endDate}
          onRangeChange={onRangeChangeHandler}
        />
      </Popover>
    );
  } else {
    return (
      <Calendar
        {...rest}
        rangePicker={true}
        startDate={startDate}
        endDate={endDate}
        onRangeChange={onRangeChange}
      />
    );
  }
};

export default RangePicker;
