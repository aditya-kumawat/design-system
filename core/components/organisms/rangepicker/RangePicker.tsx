import * as React from 'react';
import { Calendar, SharedProps } from '../calendar/Calendar';
import { DateType, DateFormat } from '../calendar/types';
import Text from '@/components/atoms/text';
import Popover, { Position } from '@/components/molecules/popover';
import InputMask, { Mask, InputMaskProps } from '@/components/molecules/inputMask';
import masks from '@/components/molecules/inputMask/masks';
import validators from '@/utils/validators';
import { getDateInfo, convertToDate, compareDate, translateToString, translateToDate, Validator } from '../calendar/utility';

export type RangePickerProps = {
  /**
   * @argument startDate Start Date object
   * @argument endDate End Date object
   * @argument startDateVal Start Date string value as per `outputFormat`
   * @argument endDateVal End Date string value as per `outputFormat`
   */
  onRangeChange?: (startDate: Date, endDate: Date, startValue?: string, endValue?: string) => void;
  /**
   * Start date of `RangePicker`
   */
  startDate?: DateType;
  /**
   * End date of `RangePicker`
   */
  endDate?: DateType;
  /**
   * Allowed limit for difference in startDate and endDate
   *
   * **set `0` or `undefined` for infinite limit**
   */
  rangeLimit?: number;
  /**
   * Set if `InputMask` should be used as trigger
   */
  withInput?: boolean;
  /**
   * Position of `RangePicker` w.r.t. `InputMask`
   */
  position?: Position;
  /**
   * Should be used if `date` is of type `string`
   * @default "mm/dd/yyyy"
   */
  inputFormat?: DateFormat;
  /**
   * Should be used to translate `date` to desired format for `onRangeChange` callback
   * @default "mm/dd/yyyy"
   */
  outputFormat?: DateFormat;
  /**
   * Separator to be used between the two `InputMask`
   * @default " - "
   */
  rangeSeparator?: string;
  /**
   * Props to be used for Start date `InputMask`
   */
  startInputProps?: Omit<InputMaskProps, 'mask' | 'value' | 'onChange' | 'Blur' | 'onClick' | 'onClear'>;
  /**
   * Props to be used for Start date `InputMask`
   */
  endInputProps?: Omit<InputMaskProps, 'mask' | 'value' | 'onChange' | 'Blur' | 'onClick' | 'onClear'>;
  /**
   * custom Mask for the mentioned inputFormat
   */
  mask?: Mask;
  /**
   * custom Validator for the mentioned inputFormat and outputFormat
   */
  validator?: Validator;
} & SharedProps;

export const RangePicker = (props: RangePickerProps) => {
  const {
    startDate: startDateProp,
    endDate: endDateProp,
    yearNav: yearNavProp,
    monthNav: monthNavProp,
    inputFormat = 'mm/dd/yyyy',
    outputFormat = 'mm/dd/yyyy',
    rangeSeparator = ' - ',
    startInputProps = {
      name: 'rangePicker-start',
      label: 'Start Date',
      placeholderChar: '_',
      placeholder: inputFormat
    },
    endInputProps = {
      name: 'rangePicker-end',
      label: 'End Date',
      placeholderChar: '_',
      placeholder: inputFormat,
    },
    mask = masks.date[inputFormat],
    validator = validators.date,
    withInput,
    position,
    disabledBefore,
    disabledAfter,
    onRangeChange,
    rangeLimit,
    ...rest
  } = props;

  const [startDate, setStartDate] = React.useState<Date | undefined>();
  const [endDate, setEndDate] = React.useState<Date | undefined>();
  const [yearNav, setYearNav] = React.useState<number | undefined>(yearNavProp);
  const [monthNav, setMonthNav] = React.useState<number | undefined>(monthNavProp);
  const [open, setOpen] = React.useState<boolean>(false);
  const [startError, setStartError] = React.useState<boolean>(false);
  const [endError, setEndError] = React.useState<boolean>(false);

  React.useEffect(() => {
    const d = startDateProp ? convertToDate(startDateProp, inputFormat, validator) : undefined;
    setStartDate(d);
  }, [startDateProp]);

  React.useEffect(() => {
    const d = endDateProp ? convertToDate(endDateProp, inputFormat, validator) : undefined;
    setEndDate(d);
  }, [endDateProp]);

  React.useEffect(() => {
    setYearNav(yearNavProp);
  }, [yearNavProp]);

  React.useEffect(() => {
    setMonthNav(monthNavProp);
  }, [monthNavProp]);

  React.useEffect(() => {
    let sError = !startDate;
    let eError = !endDate;

    const {
      year: sYear,
      month: sMonth,
    } = getDateInfo(startDate);
    const {
      year: eYear,
      month: eMonth,
      date: eDate
    } = getDateInfo(endDate);
    if (compareDate(startDate, 'more', eYear, eMonth, eDate)) {
      sError = true;
      eError = true;
    }

    if (startDate) {
      setYearNav(sYear);
      setMonthNav(sMonth);
    }
    if (endDate) {
      setYearNav(eYear);
      setMonthNav(eMonth);
    }

    setStartError(sError);
    setEndError(eError);
    if (onRangeChange) {
      if (startDate && endDate) {
        const inRangeError = getInRangeError();
        if (!inRangeError && !sError && !eError) {
          const sValue = translateToString(outputFormat, startDate);
          const eValue = translateToString(outputFormat, endDate);
          onRangeChange(startDate, endDate, sValue, eValue);
        }
      }
    }
  }, [startDate, endDate]);

  const getInRangeError = () => {
    if (rangeLimit) {
      const {
        year: sYear,
        month: sMonth,
        date: sDate
      } = getDateInfo(startDate);

      const {
        year: eYear,
        month: eMonth,
        date: eDate
      } = getDateInfo(endDate);

      let limitDate: Date;
      if (startDate) {
        limitDate = new Date(startDate);
        limitDate.setDate(sDate + rangeLimit);

        return compareDate(limitDate, 'less', eYear, eMonth, eDate + 1);
      }
      if (endDate) {
        limitDate = new Date(endDate);
        limitDate.setDate(eDate - rangeLimit);

        return compareDate(limitDate, 'more', sYear, sMonth, sDate - 1);
      }
    }
    return false;
  };

  const onRangeChangeHandler = (sDate?: Date, eDate?: Date) => {
    if (sDate) setStartDate(sDate);
    if (eDate) setEndDate(eDate);
  };

  if (withInput) {
    const updateNav = (type: string) => {
      if (type === 'start') {
        const {
          year,
          month
        } = getDateInfo(startDate);
        setYearNav(year);
        setMonthNav(month);
      }
      if (type === 'end') {
        const {
          year,
          month
        } = getDateInfo(endDate);
        setYearNav(year);
        setMonthNav(month);
      }
    };

    const onChangeHandler = (_e: React.ChangeEvent<HTMLInputElement>, val: string, type: string) => {
      setOpen(true);

      if (type === 'start') {
        const placeholderChar = startInputProps.placeholderChar ? startInputProps.placeholderChar : '_';
        if (val && !val.includes(placeholderChar)) {
          const d = translateToDate(inputFormat, val, validator);
          if (d) {
            setStartDate(d);
            if (endDate) {
              const {
                year: eYear,
                month: eMonth,
                date: eDate
              } = getDateInfo(endDate);
              if (compareDate(startDate, 'more', eYear, eMonth, eDate)) {
                setEndDate(undefined);
              }
            }
          }
        }
      }
      if (type === 'end') {
        const placeholderChar = endInputProps.placeholderChar ? endInputProps.placeholderChar : '_';
        if (val && !val.includes(placeholderChar)) {
          const d = translateToDate(inputFormat, val, validator);
          if (d) setEndDate(d);
        }
      }
    };

    const onBlurHandler = (_e: React.ChangeEvent<HTMLInputElement>, val: string, type: string) => {
      if (type === 'start') {
        const placeholderChar = startInputProps.placeholderChar ? startInputProps.placeholderChar : '_';
        if (!val || val.includes(placeholderChar)) setStartDate(undefined);
      }
      if (type === 'end') {
        const placeholderChar = endInputProps.placeholderChar ? endInputProps.placeholderChar : '_';
        if (!val || val.includes(placeholderChar)) setEndDate(undefined);
      }
    };

    const onClearHandler = (type: string) => {
      if (type === 'start') {
        setStartDate(undefined);
        updateNav('end');
      }
      if (type === 'end') {
        setEndDate(undefined);
        updateNav('start');
      }
    };

    const onClickHandler = (type: string) => {
      if (!open) {
        updateNav(type);
      }
    };

    const trigger = (
      <div className="RangePicker-input">
        <InputMask
          {...startInputProps}
          mask={mask}
          value={startDate ? translateToString(inputFormat, startDate) : ''}
          onChange={(e: React.ChangeEvent<HTMLInputElement>, val?: string) => onChangeHandler(e, val || '', 'start')}
          onBlur={(e: React.ChangeEvent<HTMLInputElement>, val?: string) => onBlurHandler(e, val || '', 'start')}
          onClear={() => onClearHandler('start')}
          onClick={() => onClickHandler('start')}
          error={startError}
        />
        <div className="RangePicker-rangeSeparator">
          <Text>{rangeSeparator}</Text>
        </div>
        <InputMask
          {...endInputProps}
          mask={mask}
          value={endDate ? translateToString(inputFormat, endDate) : ''}
          onChange={(e: React.ChangeEvent<HTMLInputElement>, val?: string) => onChangeHandler(e, val || '', 'end')}
          onBlur={(e: React.ChangeEvent<HTMLInputElement>, val?: string) => onBlurHandler(e, val || '', 'end')}
          onClear={() => onClearHandler('end')}
          onClick={() => onClickHandler('end')}
          error={endError}
        />
      </div>
    );

    const onToggleHandler = (o: boolean, type?: string) => {
      switch (type) {
        case 'outsideClick':
          setOpen(o);
          break;
        case 'onClick':
          setOpen(true);
          break;
      }
    };

    return (
      <Popover
        trigger={trigger}
        position={position}
        appendToBody={true}
        open={open}
        onToggle={onToggleHandler}
      >
        <Calendar
          {...rest}
          rangePicker={true}
          startDate={convertToDate(startDate, inputFormat, validator)}
          endDate={convertToDate(endDate, inputFormat, validator)}
          disabledBefore={convertToDate(disabledBefore, inputFormat, validator)}
          disabledAfter={convertToDate(disabledAfter, inputFormat, validator)}
          onRangeChange={onRangeChangeHandler}
          yearNav={yearNav}
          monthNav={monthNav}
          rangeLimit={rangeLimit}
        />
      </Popover>
    );
  }
  return (
    <Calendar
      {...rest}
      rangePicker={true}
      startDate={convertToDate(startDate, inputFormat, validator)}
      endDate={convertToDate(endDate, inputFormat, validator)}
      disabledBefore={convertToDate(disabledBefore, inputFormat, validator)}
      disabledAfter={convertToDate(disabledAfter, inputFormat, validator)}
      onRangeChange={onRangeChangeHandler}
      yearNav={yearNav}
      monthNav={monthNav}
      rangeLimit={rangeLimit}
    />
  );
};

RangePicker.displayName = 'RangePicker';

export default RangePicker;
