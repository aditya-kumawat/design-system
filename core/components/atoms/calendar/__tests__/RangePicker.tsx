import * as React from 'react';
import { shallow } from 'enzyme';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import RangePicker, { IRangePickerProps as IProps } from '../RangePicker';

const view = ['year', 'month', 'date'];
const booleanValue = [true, false];

describe('RangePicker component', () => {
  const mapper: Record<string, any> = {
    view: valueHelper(view, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <RangePicker
          startDate={new Date(2020, 2, 3)}
          endDate={new Date(2020, 2, 11)}
          {...attr}
        >
          RangePicker
        </RangePicker >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('RangePicker component', () => {
  const mapper: Record<string, any> = {
    monthsInView: valueHelper([1, 2, 3], { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <RangePicker
          startDate={new Date(2020, 2, 3)}
          endDate={new Date(2020, 2, 11)}
          {...attr}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('RangePicker component', () => {
  const mapper: Record<string, any> = {
    monthsInView: valueHelper([1, 2, 3], { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <RangePicker
          startDate={new Date(2020, 2, 3)}
          endDate={new Date(2020, 2, 11)}
          yearNav={2019}
          monthNav={11}
          {...attr}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('RangePicker component', () => {
  const mapper: Record<string, any> = {
    rangeLimit: valueHelper(7, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <RangePicker
          startDate={new Date(2020, 2, 3)}
          endDate={new Date(2020, 2, 11)}
          {...attr}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('RangePicker component', () => {
  const mapper: Record<string, any> = {
    withInput: valueHelper(booleanValue, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <RangePicker
          startDate={new Date(2020, 2, 3)}
          endDate={new Date(2020, 2, 11)}
          {...attr}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});
