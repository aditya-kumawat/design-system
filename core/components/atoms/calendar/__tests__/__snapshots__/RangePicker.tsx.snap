// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangePicker component 
  
 1`] = `
<Component
  onRangeChange={[Function]}
  rangePicker={true}
/>
`;

exports[`RangePicker component 
  monthsInView: 1
 1`] = `
<Component
  monthsInView={1}
  onRangeChange={[Function]}
  rangePicker={true}
/>
`;

exports[`RangePicker component 
  monthsInView: 1
 2`] = `
<Component
  monthNav={11}
  monthsInView={1}
  onRangeChange={[Function]}
  rangePicker={true}
  yearNav={2019}
/>
`;

exports[`RangePicker component 
  monthsInView: 2
 1`] = `
<Component
  monthsInView={2}
  onRangeChange={[Function]}
  rangePicker={true}
/>
`;

exports[`RangePicker component 
  monthsInView: 2
 2`] = `
<Component
  monthNav={11}
  monthsInView={2}
  onRangeChange={[Function]}
  rangePicker={true}
  yearNav={2019}
/>
`;

exports[`RangePicker component 
  monthsInView: 3
 1`] = `
<Component
  monthsInView={3}
  onRangeChange={[Function]}
  rangePicker={true}
/>
`;

exports[`RangePicker component 
  monthsInView: 3
 2`] = `
<Component
  monthNav={11}
  monthsInView={3}
  onRangeChange={[Function]}
  rangePicker={true}
  yearNav={2019}
/>
`;

exports[`RangePicker component 
  view: "date"
 1`] = `
<Component
  onRangeChange={[Function]}
  rangePicker={true}
  view="date"
>
  RangePicker
</Component>
`;

exports[`RangePicker component 
  view: "month"
 1`] = `
<Component
  onRangeChange={[Function]}
  rangePicker={true}
  view="month"
>
  RangePicker
</Component>
`;

exports[`RangePicker component 
  view: "year"
 1`] = `
<Component
  onRangeChange={[Function]}
  rangePicker={true}
  view="year"
>
  RangePicker
</Component>
`;

exports[`RangePicker component 
  withInput: false
 1`] = `
<Component
  onRangeChange={[Function]}
  rangePicker={true}
/>
`;

exports[`RangePicker component 
  withInput: true
 1`] = `
<Popover
  appendToBody={true}
  onToggle={[Function]}
  open={false}
  trigger={
    <div
      className="RangePicker-input"
    >
      <ForwardRef
        error={false}
        mask={
          Array [
            /\\\\d/,
            /\\\\d/,
            "/",
            /\\\\d/,
            /\\\\d/,
            "/",
            /\\\\d/,
            /\\\\d/,
            /\\\\d/,
            /\\\\d/,
          ]
        }
        name="rangePicker-start"
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onClick={[Function]}
        placeholder="dd/mm/yy"
        placeholderChar="_"
        value=""
      />
      <div
        className="RangePicker-rangeSeparator"
      >
        <Text>
           - 
        </Text>
      </div>
      <ForwardRef
        error={false}
        mask={
          Array [
            /\\\\d/,
            /\\\\d/,
            "/",
            /\\\\d/,
            /\\\\d/,
            "/",
            /\\\\d/,
            /\\\\d/,
            /\\\\d/,
            /\\\\d/,
          ]
        }
        name="rangePicker-end"
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onClick={[Function]}
        placeholder="dd/mm/yy"
        placeholderChar="_"
        value=""
      />
    </div>
  }
>
  <Component
    onRangeChange={[Function]}
    rangePicker={true}
  />
</Popover>
`;
